packages/docusaurus-utils/src/markdownParser.ts

```diff
@@ -86,7 +86,7 @@ export function parseMarkdownContentTitle(
 
   const content = contentUntrimmed.trim();
 
-  const regularTitleMatch = /^(?<pattern>#\s*(?<title>[^#\n]*)+\s*#*[\s\r]*?\n*?)/g.exec(
+  const regularTitleMatch = /^(?<pattern>#\s*(?<title>[^#\n]*)+[ \t]*#?\n*?)/g.exec(
```

Comment:
Thanks, that seems to work but I don't understand why 😅 can you explain me?

Also, just thought about it, but we should probably do something for the `{#headingId}` syntax, as the cli will add it to h1 headings too

Reply:
Not sure if I can do it better than just showing demo link on [regex101](https://regex101.com/r/BqBd7D/1).

We are interested in first-level heading, so next h2 with explicit id will be parsed as expected.